### PR TITLE
Improvements to pretty-printing for AND/OR/CASE

### PR DESCRIPTION
**Pull request by jpschorr**
Addresses parts of #473 :
 - Improves pretty-printing of `CASE`
 - adds round-trip testing for pretty-printing

Also:
 - Improves pretty-printing of AND/OR connectives
 - Improves pretty-printing of query clauses

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-ast/src/pretty.rs [hunk 1]
```diff
@@ -570,10 +570,15 @@ impl PrettyDoc for BinOp {
...
+        let sep = if nest == 0 {
+            arena.space()
+        } else {
+            arena.softline()
+        };
         let expr = arena.intersperse([lhs, op, rhs], sep).group();
-        let paren_expr = [arena.text("("), expr, arena.text(")")];
-        arena.concat(paren_expr).group()
+        //let paren_expr = [arena.text("("), expr, arena.text(")")];
+        //arena.concat(paren_expr).group()
```
Comment (am357) [inline partiql-ast/src/pretty.rs:580]:
```suggestion
        //let paren_expr = [arena.text("("), expr, arena.text(")")];
        //arena.concat(paren_expr).group()
```

**Review by am357 — COMMENTED** [review]
LGTM—with one suggestion to remove the lines that are commented out.

File: partiql-ast/src/pretty.rs [hunk 2]
```diff
@@ -232,25 +242,34 @@ impl PrettyDoc for Select {
...
+            delegate(where_clause, arena),
+            delegate(group_by, arena),
+            delegate(having, arena),
         ]
         .into_iter()
         .flatten();
 
-        arena.intersperse(clauses, arena.softline()).group()
+        //arena.concat(clauses).group()
+        //arena.intersperse(clauses, arena.softline()).group()
```
Comment (am357) [inline partiql-ast/src/pretty.rs:258]: Do we want to leave these here?

Comment (jpschorr) [reply partiql-ast/src/pretty.rs:258]: Nope

**Review by am357 — APPROVED** [review]

**Pull request merged**